core/common/src/types/cluster/metadata.rs

```diff
@@ -0,0 +1,191 @@
+/* Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use crate::{BytesSerializable, IggyError, types::cluster::node::ClusterNode};
+use bytes::{BufMut, Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+
+/// Represents metadata of all nodes in the cluster.
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct ClusterMetadata {
+    /// Name of the cluster.
+    pub name: String,
+    /// Unique identifier of the cluster.
+    pub id: u32,
+    /// Transport used for cluster communication (tcp, quic, http).
+    pub transport: String,
```

Comment:
Maybe let's reuse the existing `enum TransportProtocol`? It's part of the connection string module, but could be probably moved to a more generic area.